data-transfer/src/main/java/it/eng/datatransfer/service/api/DataTransferAPIService.java

```diff
@@ -354,16 +393,140 @@ public JsonNode terminateTransfer(String transferProcessId) {
 		}
 	}
 	
-	private void stateTransitionCheck (TransferState newState, TransferState currentState) {
-		if (!currentState.canTransitTo(newState)) {
-			throw new DataTransferAPIException("State transition aborted, " + currentState.name()
-					+ " state can not transition to " + newState.name());
+	public void downloadData(String transferProcessId) {
+		TransferProcess transferProcess = findTransferProcessById(transferProcessId);
+		
+		if (!transferProcess.getState().equals(TransferState.STARTED)) {
+			log.error("Download aborted, Transfer Process is not in STARTED state");
+			throw new DataTransferAPIException("Download aborted, Transfer Process is not in STARTED state");
+		}
+		
+		policyCheck(transferProcess);
+		
+		log.info("Starting download transfer process id - {} data...", transferProcessId);
+		GenericApiResponse<ExternalData> response = okHttpRestClient.downloadData(transferProcess.getDataAddress().getEndpoint(), 
+				null);
```

Comment:
In DataAddress object there can be Authorization object, with authirization string, it can be basic or bearer (see what has been done with EDC integration)
We should consider that this field MIGHT be present, and if present, pass it
DataAddress.List<EndpointProperty>
EndpointProperty.type   = authType
EndpointPorperty.value  = bearer

EndpointProperty.type = authorization (not 100% sure about this value)
EndpointProperty.value = "eyJraWQiOiJwdWJsaWMta2V5IiwiYWxnIjoiU....

https://github.com/eclipse-edc/Samples/blob/main/transfer/transfer-02-consumer-pull/README.md#3-check-the-data